### ISSUE TITLE
UPDATE on 2.0.4 does not work sometimes

### ISSUE THREAD

**Issue by rps-v** [crate/crate#5926]
**CrateDB version**:
2.0.4

**JVM version**:
openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-0ubuntu1.16.04.2-b11)
OpenJDK 64-Bit Server VM (build 25.131-b11, mixed mode)

**OS version / environment description**:
Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-47-generic x86_64)

**Problem description**:
Some UPDATE queries fail to execute (without giving errors or finishing, not even timing out !).

```sql
SELECT id as job_uuid, date_format(started) AS start, stmt FROM sys.jobs limit 100;
```

job_uuid | start | stmt
-- | -- | --
9239f730-92d7-42ec-8e06-b6ab17bff944 | 2017-07-10T23:14:14.960000Z | SELECT id as job_uuid, date_format(started) AS start, stmt FROM sys.jobs limit 100
09c7fe8f-8d02-46fe-9c91-68596c0d593a | 2017-07-10T22:49:34.316000Z | update "tracknamic"."route" set end_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
410bef51-e1ba-430a-8776-5d6da5c0cd6a | 2017-07-10T23:14:14.591000Z | SELECT table_name, schema_name, format('%s.%s', schema_name, table_name) AS fqn, _node['id'] AS node_id, state, routing_state, relocating_node, count(*), "primary", sum(num_docs), avg(num_docs), sum(size) FROM sys.shards GROUP BY table_name, schema_name, fqn, node_id, state, routing_state, relocating_node, "primary"
5e84596e-d04b-439c-8e58-98916f7fea25 | 2017-07-10T22:41:26.161000Z | update "tracknamic"."route" set start_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
672d2aec-6130-4e67-9555-dae463a64acf | 2017-07-10T22:52:19.056000Z | update "tracknamic"."route" set start_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
ddf25bc0-eb64-4fbe-adb7-d03442a213f7 | 2017-07-10T22:54:15.147000Z | update "tracknamic"."route" set end_location = 'DJ797, Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
f675706a-a292-49fb-89c1-d188caff370f | 2017-07-10T22:52:06.528000Z | update "tracknamic"."route" set start_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'

```sql
SELECT _node['name'], _node['hostname'], * FROM sys.operations limit 10;
```

_node['name'] | _node['hostname'] | id | job_id | name | started | used_bytes
-- | -- | -- | -- | -- | -- | --
s2 | s2.server.com | 0 | a73c4873-c13d-443c-a099-9dded07b1713 | collect | 1499728555487 (Mon, 10 Jul 2017 23:15:55 GMT) | NULL
s2 | s2.server.com | 0 | ddf25bc0-eb64-4fbe-adb7-d03442a213f7 | collect | 1499727255191 (Mon, 10 Jul 2017 22:54:15 GMT) | NULL
s1 | s1.server.com | 0 | a73c4873-c13d-443c-a099-9dded07b1713 | collect | 1499728555487 (Mon, 10 Jul 2017 23:15:55 GMT) | NULL
s0 | s0.server.com | 0 | a73c4873-c13d-443c-a099-9dded07b1713 | collect | 1499728555486 (Mon, 10 Jul 2017 23:15:55 GMT) | NULL
s0 | s0.server.com | 1 | a73c4873-c13d-443c-a099-9dded07b1713 | mergeOnHandler | 1499728555486 (Mon, 10 Jul 2017 23:15:55 GMT) | NULL
s0 | s0.server.com | 0 | 9e3f6aac-373f-4c37-b836-38f3e88c5ae1 | collect | 1499728555394 (Mon, 10 Jul 2017 23:15:55 GMT) | NULL
s0 | s0.server.com | 1 | ddf25bc0-eb64-4fbe-adb7-d03442a213f7 | mergeOnHandler | 1499727255147 (Mon, 10 Jul 2017 22:54:15 GMT) | NULL

**Steps to reproduce**:

Schema:
```sql
CREATE TABLE IF NOT EXISTS "tracknamic"."route" (
  ....
  "start_location" STRING,
  "end_location" STRING,
  ...
}
CLUSTERED BY ("id") INTO 6 SHARDS PARTITIONED BY ("vehicle_id", "month");
```

Ran this query : 
```sql
update "tracknamic"."route" set start_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
```
and expected it to give a result (either success or an error) but it just sits there (spinner in Admin UI, still going after **10 mins**)

Strangely, this query worked:
```sql
update "tracknamic"."route" set end_location = 'DJ797 Palota' where id = '08ffde71-b12a-4635-bba7-de0c7bdfc38a'
```
so it seems there might be a problem with that column.

I tried re-optimizing the table:
```sql
OPTIMIZE TABLE "tracknamic"."route" WITH (upgrade_segments=true);
```
and
```sql
OPTIMIZE TABLE "tracknamic"."route";
```

Now both queries fail to return a result (or even error).

Tried killing the jobs, still no effect.
```
KILL ALL;
```

**Use case**:

**Feature description**:

**Comment by joemoe** [comment]
thanks for reporting @rps-v, we'll check that.

**Comment by rps-v** [comment]
I saw some recent activity on the repo regarding stuck UPDATE queries and was wondering if there is any news on this ?

Only asking because this is affecting our production cluster and a fix would be great, as we can't downgrade - and is also affecting our users :sweat_smile:

Thanks again and keep up the good work 👍

**Comment by mfussenegger** [comment]
we were able to reproduce this and have a fix almost ready (https://github.com/crate/crate/pull/5936) it needs a bit more testing to make sure it really solves the issue. 

it will make it into 2.0.6 which will be out soon. (2.0.5 was already tagged)

**Comment by mfussenegger** [comment]
This should be fixed in 2.0.6 which has been released today. Please re-open in case you still encounter stuck updates.
